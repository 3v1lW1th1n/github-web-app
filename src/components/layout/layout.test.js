import 'jest-styled-components'

import React from 'react'
import { shallow, mount } from 'enzyme'
import { MemoryRouter } from 'react-router-dom'
import Layout from './'

describe('<Layout />', () => {
  it('should match default snapshot', () => {
    const component = shallow(<Layout title='My Page'><div>content</div></Layout>)
    expect(component).toMatchSnapshot()
  })

  it('should render the children correctly', () => {
    const component = mount(
      <MemoryRouter>
        <Layout title='My Page'>
          <div id='content'>content</div>
        </Layout>
      </MemoryRouter>
    )

    expect(component.find('div#content').length).toBe(1)
  })

  it('should have the right displayName', () => {
    expect(Layout.displayName).toEqual('Layout')
  })
})
