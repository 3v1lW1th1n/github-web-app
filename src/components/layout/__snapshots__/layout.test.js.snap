// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should match default snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Layout-nrf5sd-0",
        "isStatic": true,
        "rules": Array [
          ".layout__nav-wrapper{box-shadow:0px 0px 10px 0px rgba(0,0,0,0.75);position:fixed;width:100%;top:0;left:0;right:0;margin:0 auto;display:block;z-index:100;background-color:#fff;}.layout__nav{padding:0 16px;height:60px;display:flex;flex-direction:row;z-index:100;justify-content:space-between;overflow-x:auto;overflow-y:hidden;background-color:#fff;}.layout__nav-link{color:#000;text-decoration:none;display:flex;align-items:center;text-transform:uppercase;}.layout__nav-link + .layout__nav-link{margin-left:16px;}.layout__content{margin-top:70px;}",
        ],
      },
      "displayName": "Layout",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Layout-nrf5sd-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  title="My Page"
>
  <div>
    content
  </div>
</StyledComponent>
`;
